heat/core/types.py

```diff
@@ -185,75 +190,277 @@ class flexible(generic):
     'f':            float32,
     'f4':           float32,
     'f8':           float64,
+
+    # numpy types
+    np.bool:        bool,
+    np.uint8:       uint8,
+    np.int8:        int8,
+    np.int16:       int16,
+    np.int32:       int32,
+    np.int64:       int64,
+    np.float32:     float32,
+    np.float64:     float64,
+
+    # torch types
+    torch.uint8:    uint8,
+    torch.int8:     int8,
+    torch.int16:    int16,
+    torch.int32:    int32,
+    torch.int64:    int64,
+    torch.float32:  float32,
+    torch.float64:  float64,
+
+    # builtins
     builtins.bool:  bool,
     builtins.int:   int32,
     builtins.float: float32,
 }
 
 
-def as_torch_type(a_type):
+def canonical_heat_type(a_type):
     """
-    Infers the PyTorch type from a given HeAT type, type string or Python builtin type.
+    Canonizes the builtin Python type, type string or HeAT type into a canonical HeAT type.
```

Comment:
A really minor comment, but the literal translation of "canonizes" makes the sentence quite amusing